pkg/sql/sem/tree/cast.go

```diff
@@ -162,6 +162,14 @@ var castMap = map[oid.Oid]map[oid.Oid]cast{
 		oid.T_text:    {maxContext: CastContextAssignment, origin: contextOriginAutomaticIOConversion, volatility: VolatilityImmutable},
 		oid.T_varchar: {maxContext: CastContextAssignment, origin: contextOriginAutomaticIOConversion, volatility: VolatilityImmutable},
 	},
+	oid.T__bit: {
```

Comment:
i think PG handles array types as always stable like you say, with origin / maxcontext the same as the singular type.

at least, i can't see a pg_cast entry for these, so maybe we shouldn't either:

```
otan=# select * from pg_cast where castsource = '_timetz'::regtype::oid;
 oid | castsource | casttarget | castfunc | castcontext | castmethod
-----+------------+------------+----------+-------------+------------
(0 rows)
```

Reply:
Sorry, I'm not sure what you are suggesting I change. Can you elaborate?

Reply:
i don't think this commit needs to be in there. pg doesn't have a definition for each array type.

Reply:
There are many casts missing in `pg_cast`. Pretty much any automatic I/O conversion to/from a string type is not included. For example, there is no cast from `INT4` to `TEXT`, but the cast is allowed:

```
marcus=# SELECT * FROM pg_cast WHERE castsource = 'int4'::regtype and casttarget = 'text'::regtype;
 oid | castsource | casttarget | castfunc | castcontext | castmethod
-----+------------+------------+----------+-------------+------------
(0 rows)

marcus=# SELECT 1::INT4::TEXT;
 text
------
 1
(1 row)
```

So we need to put information about these casts somewhere, and I thought it'd be best to keep it all in the same place. The purpose of `origin: contextOriginAutomaticIOConversion` is for record-keeping: it denotes that the cast is not in PG's `pg_cast`. (But the `origin` field has no effect on behavior of the cast.)

Reply:
i guess i'm not 100% sold on them needing to be individually defined since all arrays are all stable by default and all these other fields seem inferrable by default anyway.

at the very least, it feels like they could be defined during an `init` rather than individually defined. i'm trying to avoid having to define more casts manually than we need to, which can be error prone. wdyt?

Reply:
Good point. I agree for these array->string and string->array casts. They can be implemented similar to the ENUM casts. I'll update that commit.

Reply:
Done. Let me know what you think.